packages/@aws-cdk/aws-appconfig-alpha/lib/environment.ts

```diff
@@ -274,24 +274,16 @@ export class Environment extends EnvironmentBase {
     this.application.addExistingEnvironment(this);
   }
 
-  private createAlarmRole(monitor: Monitor, index: number): iam.IRole {
-    const logicalId = monitor.isCompositeAlarm ? 'RoleCompositeAlarm' : `Role${index}`;
+  private createOrGetAlarmRole(): iam.IRole {
+    const logicalId = 'Role';
     const existingRole = this.node.tryFindChild(logicalId) as iam.IRole;
     if (existingRole) {
       return existingRole;
     }
-    const alarmArn = monitor.isCompositeAlarm
-      ? this.stack.formatArn({
-        service: 'cloudwatch',
-        resource: 'alarm',
-        resourceName: '*',
-        arnFormat: ArnFormat.COLON_RESOURCE_NAME,
-      })
-      : monitor.alarmArn;
     const policy = new iam.PolicyStatement({
       effect: iam.Effect.ALLOW,
       actions: ['cloudwatch:DescribeAlarms'],
-      resources: [alarmArn],
+      resources: ['*'],
```

Comment:
The integ test already tests this because there is only one policy being created now with resource set to `*`.

Yeah, I discussed this with a senior engineer on my team and it is only being scoped to all resources for `DescribeAlarms` so I think this should be okay. But, after doing some personal testing, I found out that the original policy we had for composite alarms actually works when scoped narrower than `*` so the CW docs must be incorrect here. We could change this to that if you want too

https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_DescribeAlarms.html

Reply:
re: integ test

the test checks if the policy is being created with the resource set to `*`. it does not yet test that that policy can be used successfully. i argue that the latter is the important part, the former is simply a unit test.

re: resource

since it is just a scope for `describeAlarms`, i.e. a readonly prop, we should be fine with `*`. Can you document in the code that we are okay with this for the readonly permissions and link to the doc page?